### DISCUSSION TITLE
Preventing Duplicate Records

### DISCUSSION THREAD

**Discussion by Arcaster42** [pocketbase/pocketbase#2217] [Q&A] [answered]
I attempted to create a rule to prevent duplicate combinations, essentially the `userId` and `topicId` combination should be unique. But I'm clearly missing something about accessing existing records to check if they exist.

![image](https://user-images.githubusercontent.com/47514505/230004506-95c7b592-770b-4244-9c87-af6ebd3a08ae.png)

How can I ensure there is always a unique `userId` to `topicId` combination?

**Comment by ganigeorgiev** [comment, marked as answer]
For the "Create rule", `userId` and `topicId` refers to the record that is being created (aka. it is identical to `@request.data.*`).

If you want to apply a unique constraint across multiple fields, you can define a unique index (this is available with v0.14+).
